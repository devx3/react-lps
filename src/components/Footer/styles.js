import styled, { css } from 'styled-components';
import { Container as TextComponent } from '../TextComponent/styles';
import { Container as SectionContainer } from '../SectionContainer/styles';

export const Container = styled.div`
    text-align: center;
    ${({ theme }) => css`
        border-top: 1px solid ${theme.colors.mediumGrey};
        a {
            text-decoration: none;
        }

        & ${TextComponent} p {
            font-size: ${theme.fonts.sizes.xsmall};
        }

        & ${SectionContainer} {
            padding-top: 0;
            padding-bottom: 0;
        }
    `};
`;
