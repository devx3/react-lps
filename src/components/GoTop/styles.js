import styled, { css } from 'styled-components';

export const Container = styled.a`
    ${({ theme }) => css`
        background: ${theme.colors.secondaryColor};
        color: ${theme.colors.primaryColor};
        display: flex;
        align-items: center;
        justify-content: center;
        position: fixed;
        width: 4rem;
        height: 4rem;
        bottom: 2rem;
        right: 2rem;
        z-index: 6;
        cursor: pointer;
    `}
`;
