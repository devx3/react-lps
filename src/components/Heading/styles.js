import styled, { css } from 'styled-components';

const titleSize = (theme, size) => {
    return css`
        font-size: ${eval(`theme.fonts.sizes.${size}`)};
        ${mediaFont(theme)}
    `;
};

const mediaFont = (theme) => css`
    @media ${theme.media.ltMedia} {
        font-size: ${theme.fonts.sizes.biggest};
    }
`;

const titleCase = (uppercase) => css`
    text-transform: ${uppercase ? 'uppercase' : 'none'};
`;

export const Title = styled.h1`
    ${({ theme, darkMode, size, uppercase }) => css`
        color: ${darkMode ? theme.colors.white : theme.colors.primaryColor};
        ${titleSize(theme, size)};
        ${titleCase(uppercase)};
    `}
`;
