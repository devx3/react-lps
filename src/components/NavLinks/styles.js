import styled, { css } from 'styled-components';

export const Container = styled.nav`
    ${({ theme }) => css`
        display: flex;
        flex-flow: row wrap;

        @media ${theme.media.ltMedia} {
            flex-flow: column wrap;
            align-content: center;
        }
    `}
`;
