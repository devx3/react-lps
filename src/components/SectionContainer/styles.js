import styled, { css } from 'styled-components';

export const Container = styled.div`
    ${({ theme }) => css`
        max-width: 120rem !important;
        margin: 0 auto;
        padding: ${theme.spacings.large};
        width: 100%;
    `}
`;
