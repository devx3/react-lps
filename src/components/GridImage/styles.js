import styled, { css } from 'styled-components';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextContainer} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};
    `}
`;

export const GridItem = styled.div`
    ${() => css`
        overflow: hidden;
    `}
`;

export const Image = styled.img`
    ${() => css`
        width: 100%;
        transition: all 300ms ease-in-out;

        &:hover {
            transform: scale(1.2) rotate(5deg);
        }
    `}
`;
