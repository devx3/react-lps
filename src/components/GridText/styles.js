import styled, { css } from 'styled-components';
import { Title as HeadingContainer } from '../Heading/styles';
import { Container as TextContainer } from '../TextComponent/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        ${TextContainer} {
            margin-bottom: ${theme.spacings.xhuge};
        }
    `}
`;

export const Grid = styled.div`
    ${({ theme }) => css`
        counter-reset: grid-counter;
        display: grid;
        grid-template-columns: repeat(auto-fill, minmax(280px, 1fr));
        gap: ${theme.spacings.large};

        @media ${theme.media.ltMedia} {
            grid-template-columns: 1fr;
        }
    `}
`;

export const GridItem = styled.div`
    ${({ theme }) => css`
        ${HeadingContainer} {
            position: relative;
            left: 5rem;
            margin-bottom: 15px;
        }

        ${HeadingContainer}::before {
            counter-increment: grid-counter;
            content: counter(grid-counter);
            position: absolute;
            top: -2rem;
            font-size: ${theme.fonts.sizes.huge};
            left: -5rem;
            transform: rotate(-10deg);
        }
    `}
`;
