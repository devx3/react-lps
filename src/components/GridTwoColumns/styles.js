import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

export const Container = styled.div`
    ${({ theme }) => css`
        display: grid;
        grid-template-columns: 1fr 2fr;
        align-items: center;
        gap: ${theme.spacings.large};

        @media ${theme.media.ltMedia} {
            grid-template-columns: 1fr;
        }
    `}
`;

export const TextContainer = styled.div`
    ${({ theme }) => css`
        @media ${theme.media.ltMedia} {
            margin-bottom: ${theme.spacings.large};
        }

        ${Heading} {
            margin-bottom: ${theme.spacings.large};
        }
    `}
`;

export const ImageContainer = styled.div`
    ${() => css``}
`;

export const Image = styled.img`
    ${() => css``}
`;
