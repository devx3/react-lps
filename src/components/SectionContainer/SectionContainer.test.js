import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { SectionContainer } from './index.jsx';

describe('<SectionContainer />', () => {
    it('Should render with default values', () => {
        const { container } = renderTheme(
            <SectionContainer>Children</SectionContainer>,
        );
        const section = screen.getByText('Children');

        expect(section).toHaveStyle('max-width: 120rem');
        expect(container).toMatchSnapshot();
    });
});
