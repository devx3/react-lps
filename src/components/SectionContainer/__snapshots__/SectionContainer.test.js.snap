// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SectionContainer /> Should render with default values 1`] = `
.c0 {
  max-width: 120rem !important;
  margin: 0 auto;
  padding: 3.2rem;
  width: 100%;
}

<div>
  <div
    class="c0"
  >
    Children
  </div>
</div>
`;
